signac/common/config.py

```diff
@@ -84,22 +83,24 @@ def load_config(root=None, local=False):
     if root is None:
         root = os.getcwd()
     config = Config(configspec=cfg.split("\n"))
-    if local:
-        for fn in _search_local(root):
-            tmp = read_config_file(fn)
-            config.merge(tmp)
-            if "project" in tmp:
-                config["project_dir"] = os.path.dirname(fn)
-                break
-    else:
+
+    if not local:
+        # For non-local searches we grab the user's global config file first.
         for fn in search_standard_dirs():
             config.merge(read_config_file(fn))
-        for fn in _search_tree(root):
-            tmp = read_config_file(fn)
-            config.merge(tmp)
-            if "project" in tmp:
-                config["project_dir"] = os.path.dirname(fn)
-                break
+        search_func = _search_tree
+    else:
+        # Local searches cannot proceed up the tree.
+        search_func = _search_local
```

Comment:
No reason. I wrote the new conditional separately since the body of the old conditional is now just the for loop below, so I didn't register the switch. Will switch back.